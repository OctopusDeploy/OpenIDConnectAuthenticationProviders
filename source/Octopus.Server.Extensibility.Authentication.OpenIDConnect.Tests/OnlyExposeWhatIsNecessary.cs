using System;
using System.Linq;
using System.Reflection;
using Assent;
using NUnit.Framework;
using Octopus.Server.Extensibility.Authentication.AzureAD;
using Octopus.Server.Extensibility.Authentication.GoogleApps;
using Octopus.Server.Extensibility.Authentication.OctopusID;
using Octopus.Server.Extensibility.Authentication.Okta;

namespace Tests
{
    [TestFixture]
    public class OnlyExposeWhatIsNecessary
    {
        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed_AzureAD()
        {
            this.Assent(CheckAssembly(typeof(AzureADExtension).Assembly));
        }

        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed_GoogleApps()
        {
            this.Assent(CheckAssembly(typeof(GoogleAppsExtension).Assembly));
        }
        
        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed_Okta()
        {
            this.Assent(CheckAssembly(typeof(OktaExtension).Assembly));
        }
        
        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed_OctopusId()
        {
            this.Assent(CheckAssembly(typeof(OctopusIDExtension).Assembly));
        }

        string CheckAssembly(Assembly assembly)
        {
            var publicThings = assembly.GetExportedTypes()
                .Select(t => t.FullName);

            return string.Join(Environment.NewLine, publicThings);
        }
    }
}