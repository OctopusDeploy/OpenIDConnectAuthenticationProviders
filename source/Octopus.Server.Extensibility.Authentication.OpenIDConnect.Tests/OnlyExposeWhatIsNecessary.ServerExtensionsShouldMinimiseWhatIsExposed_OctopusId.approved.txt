Octopus.Server.Extensibility.Authentication.OctopusID.OctopusIDExtension